odoo/addons/test_new_api/tests/test_private_fields.py

```diff
@@ -0,0 +1,21 @@
+from odoo.tests.common import TransactionCase
+
+
+class TestPrivateFields(TransactionCase):
+
+    def setUp(self):
+        self.user1 = self.env['res.users'].create({
+            'name': 'User1',
+            'login': '1',
+            'test_private_field': 'test',
+        })
+        self.user2 = self.env['res.users'].create({
+            'name': 'User2',
+            'login': '2',
+        })
+        self.env.user.flush()
+        self.env.user.invalidate_cache()
+
+    def test_private_fields(self):
+        self.assertEqual(self.user1.with_user(self.user2).test_private_field, '********')
+        self.assertEqual(self.user1.with_user(self.user1).test_private_field, 'test')
```

Comment:
For this one to work, you can invalidate the cache before accessing the field again
```py
self.user1.invalidate_cache(['test_private_field'])
```